### ISSUE TITLE
race: can't detect race when value is set via reflect

### ISSUE THREAD

**Issue by jbardin** [golang/go#16946]
go version 1.6.3, 1.7 on  darwin/amd64 and linux/amd64

The following code does not trigger the race detector
```
func main() {
	var j int
	for i := 0; i < 10000; i++ {
		go reflectSet(&j, i)
		fmt.Println(j)
	}
}

func reflectSet(to, from interface{}) {
	reflect.ValueOf(to).Elem().Set(reflect.ValueOf(from))
}
```

### What did you expect to see?

`WARNING: DATA RACE`

### What did you see instead?

No race was detected. The size of the value being set doesn't seem to matter, so I used `int` here for brevity.

**Comment by freeformz** [comment]
Dupe of #16281 ?

**Comment by bradfitz** [comment]
Yeah, looks like this was fixed Aug 23rd, after Go 1.7 was out.

**Comment by jbardin** [comment]
@freeformz, thanks, could find that one.
